src/api/procedures/setCustomPermissions.ts

```diff
@@ -0,0 +1,100 @@
+import { isEqual } from 'lodash';
+
+import { CustomPermissionGroup } from '~/api/entities/CustomPermissionGroup';
+import { SecurityToken } from '~/api/entities/SecurityToken';
+import { PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, TransactionPermissions, TxGroup, TxTags } from '~/types';
+import { ProcedureAuthorization } from '~/types/internal';
+import {
+  numberToU32,
+  permissionsLikeToPermissions,
+  stringToTicker,
+  transactionPermissionsToExtrinsicPermissions,
+} from '~/utils/conversion';
+
+export interface SetCustomPermissionsParams {
+  permissions:
+    | {
+        transactions: TransactionPermissions;
+      }
+    | {
+        transactionGroups: TxGroup[];
+      };
+}
+
+/**
+ * @hidden
+ */
+export type Params = { group: CustomPermissionGroup } & SetCustomPermissionsParams;
+
+/**
+ * @hidden
+ */
+export async function prepareSetCustomPermissions(
+  this: Procedure<Params, void>,
+  args: Params
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: {
+        tx: { externalAgents },
+      },
+    },
+    context,
+  } = this;
+
+  const { group, permissions } = args;
+
+  const { transactions } = permissionsLikeToPermissions(permissions, context);
+  const sortedTransactions = transactions && {
+    ...transactions,
+    values: [...transactions.values].sort(),
```

Comment:
Instead of sorting here, `permissionsLikeToPermissions` should return the transactions sorted (be careful to use the spread operator there as well to avoid mutating the input). Once you do that, please remove the sorting from `createGroup` as well